source/game.c

```diff
@@ -2351,6 +2400,36 @@ static void played_cards_update_loop(bool* discarded_card, int* played_selection
     }
 }
 
+// update flames animation every 5 frames for 12FPS
+#define SCORE_FLAMES_ANIM_FREQ 5
+
+static void process_flaming_score()
+{
+    static u8 flame_score_chips_frame = 0;
+    static u8 flame_score_mult_frame  = NB_SCORE_FLAMES_FRAMES / 2; // offset mult flames animation by half the number of frames for them to not play in sync
+    
+    if (score_flames_active)
+    {
+		// animate flames at 12FPS = change sprites every 5 frames
+        if (timer % SCORE_FLAMES_ANIM_FREQ == 0)
+        {
+            // cycling through 8 frames total
+            flame_score_chips_frame = (flame_score_chips_frame + 1) % NB_SCORE_FLAMES_FRAMES;
+            flame_score_mult_frame  = (flame_score_mult_frame  + 1) % NB_SCORE_FLAMES_FRAMES;
+            // chips flame (blue)
+            main_bg_se_copy_rect(score_flame_chips_frames[flame_score_chips_frame + 1], score_flame_chips);
+            // mult flame (red)
+            main_bg_se_copy_rect(score_flame_mult_frames [flame_score_mult_frame  + 1], score_flame_mult);
+        }
```

Comment:
```suggestion
            Rect curr_frame_chips_rect = SCORE_FLAMES_CHIPS_FRAME_RECT;
            flame_score_chips_frame_idx = (flame_score_chips_frame + 1) % NB_SCORE_FLAMES_FRAMES;
            flame_score_mult_frame_idx  = (flame_score_mult_frame  + 1) % NB_SCORE_FLAMES_FRAMES;
            
            curr_frame_chips_rect.top += flame_score_chips_frame_idx;
            curr_frame_chips_rect.bottom += flame_score_chips_frame_idx;
            
            // chips flame (blue)
            main_bg_se_copy_rect(curr_frame_chips_rect, score_flame_chips);
            // mult flame (red)
            main_bg_se_copy_rect(score_flame_mult_frames [flame_score_mult_frame  + 1], score_flame_mult);
        }
```

This suggestion is not a full rewrite, more of a sketch of how it would look.

Reply:
I did basically this, but reused the frame rect variable by shifting it by 3 tiles to the right to fetch the red flames. Also shifted the frames themselves by 4 so I can use the same frame index for both animations. This all condenses the needed values to 2 Rects and 2 Points